src/Type/FloatType.php

```diff
@@ -201,6 +201,12 @@ public function traverse(callable $cb): Type
 		return $this;
 	}
 
+	public function exponentiate(Type $exponent): Type
+	{
+		$helper = new ExponentiateHelper();
+		return $helper->exponentiate($this, $exponent);
```

Comment:
It's weird to `new` this. Typically we inject these services through DI. But I understand that we're in a value object where DI is not accessible. In these cases I'd write this as a static method: `ExponentiateHelper::exponentiate`. Thanks.

Reply:
good point, fixed.